static_provider_test.go

```diff
@@ -343,3 +343,29 @@ func TestConfigDefaultsAreOverriddenByHigherPriorityProviders(t *testing.T) {
 	)
 	assert.Equal(t, book{Title: "The Financier", Author: "Dreiser", Year: 1925}, novel)
 }
+
+func TestMissingValuesForTryMethods(t *testing.T) {
+	t.Parallel()
+
+	assert := assert.New(t)
+	p := NewStaticProvider(nil)
+	t.Run("int", func(t *testing.T) {
+		_, ok := p.Get("missing").TryAsInt()
+		assert.False(ok)
+	})
+
+	t.Run("float", func(t *testing.T) {
```

Comment:
table-driven as opposed to sub-test would be much easier to write and maintain I think, but it's up to you